src/mongocrypt.h

```diff
@@ -69,8 +69,14 @@ const char *mongocrypt_version(uint32_t *len);
  * mongocrypt_ctx_mongo_op guarantees that the viewed data of
  * mongocrypt_binary_t is valid until the parent ctx is destroyed with @ref
  * mongocrypt_ctx_destroy.
+ *
+ * The `mongocrypt_binary_t` struct definition is public.
+ * Consumers may rely on the struct layout.
  */
-typedef struct _mongocrypt_binary_t mongocrypt_binary_t;
+typedef struct _mongocrypt_binary_t {
+    uint8_t *data;
+    uint32_t len;
```

Comment:
```suggestion
    void *data;
    size_t len;
```
Noting that making this public marks a point of no return: the opportunity to change these data members without requiring a subsequent API/ABI break is now. Is there compelling reason/rationale to preserve `uint32_t` as the type of `len` rather than using `size_t` or `uint64_t`?

The `uint8_t*` -> `void*` recommendation is just for better consistency with C convention for a "pointer to a blob of bytes" (i.e. see C stdlib functions such as `memcpy()`, which also saves users an explicit cast to `uint8_t*` from their own pointer-to-blob types such as `char*`). AFAIK it should make no functional difference, although it may complicate our implementation by requiring a cast back to `uint8_t*` before performing pointer arithmetic, which atm seems to only impact [`_crypto_aes_256_ctr_encrypt_decrypt_via_ecb()`](https://github.com/mongodb/libmongocrypt/blob/ce14f0cebfe9d33bf5d2f5467a0fb8a36908e80c/src/mongocrypt-crypto.c#L81) and some test code, sans any potential external uses of `mongocrypt-binary-private.h`.